### PR TITLE
Change current network protocol version for NU5 testnet

### LINKED ISSUE #2794: Set the CURRENT_NETWORK_PROTOCOL_VERSION to 170_014 for NU5 testnet
See #1841 for details.

### Tasks

  - Remove `unimplemented!("V5 transaction validation is not yet complete");` from the transaction verifier

### PR DESCRIPTION
**Pull request by upbqdn**
This PR will close #2794.

### CODE REVIEW

File: zebra-consensus/src/transaction.rs [hunk 1]
```diff
@@ -366,29 +366,29 @@ where
...
+        )
 
         // TODO:
         // - verify orchard shielded pool (ZIP-224) (#2105)
         // - ZIP-216 (#1798)
         // - ZIP-244 (#1874)
         // - remaining consensus rules (#2379)
         // - remove `should_panic` from tests
-
-        unimplemented!("V5 transaction validation is not yet complete");
```
Comment (teor2345) [inline zebra-consensus/src/transaction.rs:391]: I think this change will conflict with @conradoplg's PR #2802.

We should put these changes in whichever PR actually needs them to pass the tests.
(Or just merge this PR, and rebase PR #2802.)

File: zebra-network/src/constants.rs [hunk 2]
```diff
@@ -116,7 +116,7 @@ pub const USER_AGENT: &str = "/Zebra:1.0.0-alpha.17/";
 ///
 /// The current protocol version typically changes before Mainnet and Testnet
 /// network upgrades.
-pub const CURRENT_NETWORK_PROTOCOL_VERSION: Version = Version(170_013);
+pub const CURRENT_NETWORK_PROTOCOL_VERSION: Version = Version(170_014);
```
Comment (teor2345) [inline zebra-network/src/constants.rs:119]: I double-checked that this matches the testnet network protocol version in https://github.com/zcash/zips/blob/main/zip-0252.rst#nu5-deployment

**Review by teor2345 — APPROVED** [review]
Looks good, let's merge unless @conradoplg wants to resolve the merge conflict a different way.

File: zebra-consensus/src/transaction.rs (hunk 1 above)
Comment (conradoplg) [reply zebra-consensus/src/transaction.rs:391]: I'll rebase, should be simple

**Pull request merged**